library/cartridge_patch_instance_in_runtime.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python
+
+from copy import deepcopy
+
+from ansible.module_utils.helpers import Helpers as helpers
+
+argument_spec = {
+    'console_sock': {'required': True, 'type': 'str'},
+    'instance_config': {'required': True, 'type': 'dict'},
+    'cartridge_defaults': {'required': False, 'type': 'dict', 'default': {}},
+    'strict_mode': {'required': False, 'type': 'bool', 'default': False},
+}
+
+
+def change_memory_size(control_console, param_name, new_value):
+    if new_value is None:
+        return False, None
+
+    increase_memory_func_body = '''
+        local param_name, new_memory_size = ...
+        local ok, err = pcall(box.cfg, { [param_name] = new_memory_size })
+        if not ok then
+            if tostring(err):find("cannot decrease memory size at runtime") == nil then
+                return nil, string.format('failed to set %s: %s', param_name, err)
+            end
+        end
+        return ok
+    '''
+    return control_console.eval_res_err(increase_memory_func_body, param_name, new_value)
+
+
+def change_dynamic_params(control_console, old_box_config, new_config, strict_mode=False):
+    memory_sizes_to_change = {}
+    box_params_to_change = {}
+    incorrect_memory_changes = []
+    non_dynamic_params = []
+
+    for param_name, new_value in new_config.items():
+        old_box_value = old_box_config.get(param_name)
+
+        if param_name in helpers.MEMORY_SIZE_BOX_CFG_PARAMS:
+            if new_value > old_box_value:
+                memory_sizes_to_change.update({param_name: new_value})
+            else:
+                incorrect_memory_changes.append((param_name, old_box_value, new_value))
+
+        elif param_name in helpers.DYNAMIC_BOX_CFG_PARAMS:
+            if new_value != old_box_value:
+                box_params_to_change.update({param_name: new_value})
+
+        else:
+            non_dynamic_params.append(param_name)
+
+    if strict_mode:
+        messages = []
+        if incorrect_memory_changes:
+            messages.append('impossible to decrease memory sizes in runtime (%s)' % ', '.join(
+                map(lambda change_info: "'%s' from '%s' to '%s'" % change_info, incorrect_memory_changes)
+            ))
+        if non_dynamic_params:
+            messages.append("impossible to change '%s' in runtime" % ', '.join(non_dynamic_params))
+        if messages:
+            return None, '; '.join(messages)
```

Comment:
The result error message will start with a lowercase letter.

Reply:
What about `"Errors occurred on applying params: %s" % '';".join(messages)` or smth like this?